kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/validation/JsonSyntaxRecordValidationIT.java

```diff
@@ -5,229 +5,233 @@
-                                Map.of("allowsNulls", true, "syntacticallyCorrectJson", Map.of("validateObjectKeysUnique", true)))))
-                        .build());
+                .addToFilters(createFilterDef(false, topic));
         try (var tester = kroxyliciousTester(config);
-                var producer = getProducer(tester, 0, 16384)) {
-            Future<RecordMetadata> invalid = producer.send(new ProducerRecord<>(TOPIC_1, "my-key", SYNTACTICALLY_INCORRECT_JSON));
-            assertInvalidRecordExceptionThrown(invalid, "value was not syntactically correct JSON");
+                var producer = tester.producer()) {
+            var invalid = producer.send(new ProducerRecord<>(topic.name(), "my-key", SYNTACTICALLY_INCORRECT_JSON));
+            assertThatFutureFails(invalid, InvalidRecordException.class, "value was not syntactically correct JSON");
         }
     }
 
     @Test
-    void testInvalidJsonProduceRejectedUsingTopicNames(KafkaCluster cluster, Admin admin) throws Exception {
+    void invalidJsonProduceRejectedUsingTopicNames(KafkaCluster cluster, Topic topic1, Topic topic2) {
         assertThat(cluster.getNumOfBrokers()).isOne();
-        createTopics(admin, new NewTopic(TOPIC_1, 1, (short) 1), new NewTopic(TOPIC_2, 1, (short) 1));
 
         var config = proxy(cluster)
-                .addToFilters(new FilterDefinitionBuilder(RecordValidation.class.getName()).withConfig("rules",
-                        List.of(Map.of("topicNames", List.of(TOPIC_1), "valueRule",
-                                Map.of("allowsNulls", true, "syntacticallyCorrectJson", Map.of("validateObjectKeysUnique", true)))))
-                        .build());
+                .addToFilters(createFilterDef(false, topic1));
         try (var tester = kroxyliciousTester(config);
-                var producer = getProducer(tester, 0, 16384);
-                var consumer = getConsumer(tester)) {
-            Future<RecordMetadata> invalid = producer.send(new ProducerRecord<>(TOPIC_1, "my-key", SYNTACTICALLY_INCORRECT_JSON));
-            assertInvalidRecordExceptionThrown(invalid, "value was not syntactically correct JSON");
-            producer.send(new ProducerRecord<>(TOPIC_2, "my-key", SYNTACTICALLY_INCORRECT_JSON)).get();
-            consumer.subscribe(Set.of(TOPIC_2));
-            var records = consumer.poll(Duration.ofSeconds(10));
-            assertThat(records.count()).isOne();
-            assertThat(records.iterator().next().value()).isEqualTo(SYNTACTICALLY_INCORRECT_JSON);
+                var producer = tester.producer()) {
+            var rejected = producer.send(new ProducerRecord<>(topic1.name(), "my-key", SYNTACTICALLY_INCORRECT_JSON));
+            assertThatFutureFails(rejected, InvalidRecordException.class, "value was not syntactically correct JSON");
+
+            var accepted = producer.send(new ProducerRecord<>(topic2.name(), "my-key", SYNTACTICALLY_INCORRECT_JSON));
+            assertThatFutureSucceeds(accepted);
```

Comment:
Struggling a bit with seeing a second assertion, particularly one that succeeds in a tests called `invalidJsonProduceRejectedUsingTopicNames` but this is just a refactoring so lets leave it.